### PR TITLE
Add SymInt support

### LINKED ISSUE #50: deferred_init on HF models regression: `aten::expand` has an argument of type `SymInt` which is not supported in a deferred-init context
Many HF models stopped working:

```
  File "/home/pbelevich/local/PiPPy/examples/slurm/hf/bert/pippy_bert.py", line 102, in run_master
    bert = deferred_init(BertLMHeadModel, BertConfig(is_decoder=True))  # bert = BertLMHeadModel(BertConfig(is_decoder=True))
  File "/home/pbelevich/local/anaconda3/envs/pippy/lib/python3.9/site-packages/torchdistx/deferred_init.py", line 40, in deferred_init
    return module_fn(*args, **kwargs)
  File "/home/pbelevich/local/anaconda3/envs/pippy/lib/python3.9/site-packages/transformers/models/bert/modeling_bert.py", line 1168, in __init__
    self.bert = BertModel(config, add_pooling_layer=False)
  File "/home/pbelevich/local/anaconda3/envs/pippy/lib/python3.9/site-packages/transformers/models/bert/modeling_bert.py", line 887, in __init__
    self.embeddings = BertEmbeddings(config)
  File "/home/pbelevich/local/anaconda3/envs/pippy/lib/python3.9/site-packages/transformers/models/bert/modeling_bert.py", line 197, in __init__
    self.register_buffer("position_ids", torch.arange(config.max_position_embeddings).expand((1, -1)))
RuntimeError: `aten::expand` has an argument of type `SymInt` which is not supported in a deferred-init context.
```

Also affected models: AlbertModel, BertModel, BigBirdModel, CanineModel, CLIPModel, ConvBertModel, DebertaModel, DistilBertModel, ElectraModel, FlaubertModel, FNetModel, IBertModel, LayoutLMModel, MegatronBertModel, MobileBertModel, MPNetModel, HubertModel, RemBertModel, RetriBertModel, RobertaModel, SEWModel, SEWDModel, SplinterModel, SqueezeBertModel, ...

[test script](https://gist.github.com/pbelevich/18f1ed7d3fdce77eaf58cf239f29ddb6)

@jamesr66a @kwen2501 @Chillee @rohan-varma @H-Huang

### PR DESCRIPTION
**Pull request by pbelevich**
Fixes #50

### CODE REVIEW

File: src/cc/torchdistx/deferred_init.cc [hunk 1]
```diff
@@ -240,7 +240,8 @@ void Op::validateStack(const Stack& s) const {
                 value.isNone() ||
                 value.isString() ||
                 value.isTensor() ||
-                value.isTuple(),
+                value.isTuple() ||
+                value.isSymInt(),
```
Comment (H-Huang) [inline src/cc/torchdistx/deferred_init.cc:244]: I assume SymInt is an immutable type as the comment mentions above?

**Review by H-Huang — APPROVED** [review]
LGTM!

**Review by rohan-varma — APPROVED** [review]

File: src/cc/torchdistx/deferred_init.cc (hunk 1 above)
Comment (pbelevich) [reply src/cc/torchdistx/deferred_init.cc:244]: yeah, @Krovatkin confirmed

**Comment by cbalioglu** [comment]
Thanks for the fix @pbelevich!

**Pull request merged**